chef_actions.go

```diff
@@ -0,0 +1,239 @@
+//
+// Copyright:: Copyright 2017 Chef Software, Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+var entityNameList = map[int]string{
+	0: "nginx",
+	1: "apache",
+	2: "burger",
+	3: "salsa",
+}
+
+var requestorNameList = map[int]string{
+	0: "kyleen",
+	1: "localhost", // This is a chef-zero run
+	2: "knife",
+	3: "rad",
+	4: "lance",
+	5: "afiune",
+}
+
+var serviceHostnameList = map[int]string{
+	0: "hostname",
+	1: "localhost",
+	2: "chef.example.com",
+	3: "my.awesome.hostname.com",
+}
+
+var organizationNameList = map[int]string{
+	0: "awesome",
+	1: "_default",
+	2: "development",
+	3: "staging",
+	4: "production",
+}
+
+// ActionType will be our enum to identity a list of actions types
+type ActionType int
+
+// Supported Action Types (EntityType)
+const (
+	nodeAction ActionType = iota
+	cookbookAction
+	dataBagAction
+	environmentAction
+	roleAction
+	policyfileAction
+	// TODO: (@afiune) Add latter when compliance joins the pool party
+	//profileAction
+	//scanJobsAction
+)
+
+// Strings of the supported Action Type list above
+var actionTypeString = map[ActionType]string{
+	nodeAction:        "node",
+	cookbookAction:    "cookbook",
+	dataBagAction:     "bag",
+	environmentAction: "environment",
+	roleAction:        "role",
+	policyfileAction:  "policyfile",
+
+	// In the RFC we have more action/entity types
+	// https://github.com/chef/chef-rfc/blob/master/rfc077-mode-agnostic-data-collection.md#action-schema
+	// TODO: (@afiune) How do we handle those?
+	//"user",
+	//"permission",
+	//"item",
+	//"group",
+	//"client",
+
+	// TODO: (@afiune) Add latter when compliance joins the pool party
+	//profileAction:     "profile",
+	//scanJobsAction:    "scanjobs",
+}
+
+// Task will be our enum to identity a list of tasks
+type Task int
+
+// Supported Tasks
+const (
+	createTask Task = iota
+	editTask
+	deleteTask
+)
+
+// Strings of the supported Tasks list above
+var tasksString = map[Task]string{
+	createTask: "create",
+	editTask:   "edit",
+	deleteTask: "delete",
+}
+
+type actionRequest struct {
+	ID               uuid.UUID   `json:"id"`
+	MessageType      string      `json:"message_type"`
+	MessageVersion   string      `json:"message_version"`
+	EntityType       string      `json:"entity_type"`
+	actionType       ActionType  `json:"-"`
```

Comment:
Should that be uppercase `ActionType` in the variable name?

Reply:
Nope, it's an internal field to have an Enum for the EntityType.